ghcide/src/Development/IDE/Core/Service.hs

```diff
@@ -37,26 +37,18 @@ import Control.Monad
 -- Exposed API
 
 -- | Initialise the Compiler Service.
-initialise :: LSP.ClientCapabilities
-           -> Rules ()
-           -> IO LSP.LspId
-           -> (LSP.FromServerMessage -> IO ())
-           -> WithProgressFunc
-           -> WithIndefiniteProgressFunc
+initialise :: Rules ()
+           -> Maybe (LSP.LanguageContextEnv Config)
```

Comment:
Hiding the LSP functionality into LanguageContextEnv is exactly what I had in mind 👍